### PR TITLE
MON-2091: Add support for user-defined alert relabel configs

### PR DESCRIPTION
**Pull request by jan--f**
* [ ] TODO: Add CHANGELOG entry for this change.

### CODE REVIEW

**Comment by jan--f** [comment]
/label docs-approved
Adding doc approval as per agreement.
Ping @juzhao for qe approval.
Ping @Senthamilarasu-STA for px approval.

**Review by simonpasquier — APPROVED** [review]
/lgtm

**Comment by juzhao** [comment]
tested with PR
`launch 4.11.0-0.ci-2022-06-17-054342,openshift/cluster-monitoring-operator#1676 aws`
enabled UWM and TechPreviewNoUpgrade featuregate, can not create AlertRelabelConfig resource due to there is AlertRelabelConfig crd
```
# oc get featuregate/cluster -oyaml
apiVersion: config.openshift.io/v1
kind: FeatureGate
metadata:
  annotations:
    include.release.openshift.io/ibm-cloud-managed: "true"
    include.release.openshift.io/self-managed-high-availability: "true"
    include.release.openshift.io/single-node-developer: "true"
    release.openshift.io/create-only: "true"
  creationTimestamp: "2022-06-17T13:23:53Z"
  generation: 2
  name: cluster
  ownerReferences:
  - apiVersion: config.openshift.io/v1
    kind: ClusterVersion
    name: version
    uid: 09293241-2012-49e0-a736-62cdf2c2b100
  resourceVersion: "30481"
  uid: e8a9a154-4d84-431f-8c79-898282b117db
spec:
  featureSet: TechPreviewNoUpgrade
```
```
# oc create -f - << EOF
apiVersion: monitoring.openshift.io/v1alpha1
kind: AlertRelabelConfig
metadata:
  name: watchdog
  namespace: openshift-monitoring
spec:
  configs:
  - sourceLabels: [alertname, severity]
    regex: "Watchdog;none"
    targetLabel: severity
    replacement: critical
    action: replace
EOF
error: resource mapping not found for name: "watchdog" namespace: "openshift-monitoring" from "STDIN": no matches for kind "AlertRelabelConfig" in version "monitoring.openshift.io/v1alpha1"
ensure CRDs are installed first
```
no AlertRelabelConfig crd
```
# oc get crd | grep -i monitoring
alertmanagerconfigs.monitoring.coreos.com                         2022-06-17T13:23:29Z
alertmanagers.monitoring.coreos.com                               2022-06-17T13:23:32Z
podmonitors.monitoring.coreos.com                                 2022-06-17T13:23:34Z
probes.monitoring.coreos.com                                      2022-06-17T13:23:37Z
prometheuses.monitoring.coreos.com                                2022-06-17T13:23:40Z
prometheusrules.monitoring.coreos.com                             2022-06-17T13:23:41Z
servicemonitors.monitoring.coreos.com                             2022-06-17T13:23:44Z
thanosrulers.monitoring.coreos.com                                2022-06-17T13:23:47Z
```

**Comment by simonpasquier** [comment]
/hold

until we verify that https://github.com/openshift/cluster-monitoring-operator/pull/1675 includes the new CRD enabling the `TechPreviewNoUpgrade` feature set.

File: examples/alert-relabel-config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: watchdog
+  namespace: openshift-monitoring
+spec:
+  configs:
+  - sourceLabels: [alertname, severity]
+    regex: "Watchdog;none"
+    targetLabel: severity
+    replacement: critical
+    action: replace
```
Comment (juzhao) [inline examples/alert-relabel-config.yaml:13]:
```
# oc create -f - << EOF
> apiVersion: monitoring.openshift.io/v1alpha1
> kind: AlertRelabelConfig
> metadata:
>   name: watchdog
>   namespace: openshift-monitoring
> spec:
>   configs:
>   - sourceLabels: [alertname, severity]
>     regex: "Watchdog;none"
>     targetLabel: severity
>     replacement: critical
>     action: replace
> EOF
The AlertRelabelConfig "watchdog" is invalid: spec.configs[0].action: Unsupported value: "replace": supported values: "Replace", "Keep", "Drop", "HashMod", "LabelMap", "LabelDrop", "LabelKeep"
```

Comment (juzhao) [inline examples/alert-relabel-config.yaml:13]: see bug: https://bugzilla.redhat.com/show_bug.cgi?id=2100028

**Comment by juzhao** [comment]
maybe an RFE: https://bugzilla.redhat.com/show_bug.cgi?id=2100312
it does not block the test, and passed testing
/label qe-approved

**Pull request merged**